client/my-sites/checkout/composite-checkout/wpcom/components/wp-checkout-order-review.js

```diff
@@ -23,13 +25,20 @@ export default function WPCheckoutOrderReview( {
 	variantSelectOverride,
 	getItemVariants,
 	onChangePlanLength,
+	siteUrl,
 } ) {
+	const translate = useTranslate();
 	const [ items, total ] = useLineItems();
 	const { formStatus } = useFormStatus();
 	const isPurchaseFree = total.amount.value === 0;
 
+	const firstDomainItem = items.find( isLineItemADomain );
+	const domainUrl = firstDomainItem ? firstDomainItem.sublabel : siteUrl;
+
 	return (
 		<div className={ joinClasses( [ className, 'checkout-review-order' ] ) }>
+			{ domainUrl && <DomainURL>{ translate( 'Site' ) + ': ' + domainUrl }</DomainURL> }
```

Comment:
I have some concerns about how other locales (particularly RTL languages) might handle this translation (we should probably include the url as a replacement string in the translation). Even though we could probably wait until the i18n phase, it's probably better to do this now. See [here](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/checkout/composite-checkout/wpcom/components/wp-order-review-line-items.js#L229) for an example.